client/homescreen/activity-panel/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { withSelect } from '@wordpress/data';
+import { Card, Panel, PanelBody, PanelRow } from '@wordpress/components';
+import { more } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getUnreadOrders } from './orders/utils';
+import { getAllPanels } from './panels';
+
+const ActivityPanel = ( { panels } ) => {
+	const getTitleAndCount = ( title, count ) => {
+		return (
+			<span className="woocommerce-activity-panel-header">
+				<span className="woocommerce-activity-panel-title">
+					{ title }
+				</span>
+				{ count !== null && (
+					<span className="woocommerce-activity-panel-badge">
+						{ count }
+					</span>
+				) }
+			</span>
+		);
+	};
+
+	const renderPanels = () => {
+		return panels.map( ( panel, index ) => {
+			const { count, title, initialOpen, panel: panelContent } = panel;
+			return (
+				<Card
+					key={ index }
+					size="large"
+					className="woocommerce-activity-panel-card woocommerce-homescreen-card"
+				>
+					<PanelBody
+						title={ getTitleAndCount( title, count ) }
+						icon={ more }
+						initialOpen={ initialOpen }
+					>
+						<PanelRow> { panelContent } </PanelRow>
+					</PanelBody>
+				</Card>
+			);
+		} );
+	};
+
+	return <Panel> { renderPanels() } </Panel>;
+};
+
+export default withSelect( ( select ) => {
```

Comment:
So i mentioned some code smells with this component in the tests but if you change those things this should also change.

If you want to pass state into `ActivityPanel` you will be able to pass it as props from its parent which is better because we separate the part thats just rendering from the part that should know about state.

Reply:
One other comment, I think we should consider that any HOC that gets state should use hooks from now on rather than using `compose` and `withSelect`. It makes sense to choose one approach and stick to it for consistency.

Reply:
> One other comment, I think we should consider that any HOC that gets state should use hooks from now on rather than using compose and withSelect. It makes sense to choose one approach and stick to it for consistency.

This was modified in the commit a0c23d2